src/device.rs

```diff
@@ -384,8 +383,14 @@ impl<T: AsFd> Device<T> {
         offsets: [i32; 4],
         modifier: Modifier,
     ) -> IoResult<BufferObject<U>> {
+        let fds = [
+            buffers[0].as_raw_fd(),
+            buffers[1].as_raw_fd(),
+            buffers[2].as_raw_fd(),
+            buffers[3].as_raw_fd(),
+        ];
```

Comment:
Fwiw `BorrowedFd` is `repr(transparent)` to have the same layout as `RawFd`.

Reply:
Yeah, this should be equivalent to an `unsafe` `mem::transmute`. The compiler hopefully will recognize this and generate the same code. It could be changed, but my inclination is generally to stick with safe code.

It would also be possible to define `gbm_import_fd_modifier_data` to use a lifetime bound and `BorrowedFd`, but that's not so practical when the struct is in a generated file.

Reply:
Oh, actually, I suppose this should accept `[Option<BorrowedFd<'_>>; 4]`, or you can't pass less than four file descriptors. I kind of wish something like `arrayvec` were in `std`...

Should still transmute, with `None` mapping to `-1`, I think. Presumably passing -1 in place of an fd to GBM will return an error, but is safe.

I guess a more idiomatic way to define something like `import_buffer_object_from_dma_buf_with_modifiers` would be a builder pattern with a method to add planes? And an error if you add more than 4.

Reply:
Sounds good, and looks like you also discovered that `.map()` on a sized array gives back an array of the same size :+1: